src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -1101,222 +406,211 @@ namespace JSONHelpers
         }
     }
 
-    json::JsonObject CanvasLayoutInfo::ToJson(const CanvasLayoutInfo& canvasInfo)
+    json::JsonObject GetPersistFancyZonesJSON(const std::wstring& zonesSettingsFilePath, const std::wstring& appZoneHistoryFilePath)
     {
-        json::JsonObject infoJson{};
-        infoJson.SetNamedValue(L"ref-width", json::value(canvasInfo.lastWorkAreaWidth));
-        infoJson.SetNamedValue(L"ref-height", json::value(canvasInfo.lastWorkAreaHeight));
-
-        json::JsonArray zonesJson;
-
-        for (const auto& [x, y, width, height] : canvasInfo.zones)
+        auto result = json::from_file(zonesSettingsFilePath);
+        if (result)
         {
-            json::JsonObject zoneJson;
-            zoneJson.SetNamedValue(L"X", json::value(x));
-            zoneJson.SetNamedValue(L"Y", json::value(y));
-            zoneJson.SetNamedValue(L"width", json::value(width));
-            zoneJson.SetNamedValue(L"height", json::value(height));
-            zonesJson.Append(zoneJson);
+            if (!result->HasKey(L"app-zone-history"))
+            {
+                auto appZoneHistory = json::from_file(appZoneHistoryFilePath);
+                if (appZoneHistory)
+                {
+                    result->SetNamedValue(L"app-zone-history", appZoneHistory->GetNamedArray(L"app-zone-history"));
+                }
+                else
+                {
+                    result->SetNamedValue(L"app-zone-history", json::JsonArray());
+                }
+            }
+            return *result;
+        }
+        else
+        {
+            return json::JsonObject();
         }
-        infoJson.SetNamedValue(L"zones", zonesJson);
-        return infoJson;
     }
 
-    std::optional<CanvasLayoutInfo> CanvasLayoutInfo::FromJson(const json::JsonObject& infoJson)
+    TAppZoneHistoryMap ParseAppZoneHistory(const json::JsonObject& fancyZonesDataJSON)
     {
         try
         {
-            CanvasLayoutInfo info;
-            info.lastWorkAreaWidth = static_cast<int>(infoJson.GetNamedNumber(L"ref-width"));
-            info.lastWorkAreaHeight = static_cast<int>(infoJson.GetNamedNumber(L"ref-height"));
+            TAppZoneHistoryMap appZoneHistoryMap{};
+            auto appLastZones = fancyZonesDataJSON.GetNamedArray(L"app-zone-history");
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.